packages/builder/src/components/design/settings/controls/FieldConfiguration/PrimaryColumnFieldSetting.svelte

```diff
@@ -0,0 +1,96 @@
+<script>
+  import EditComponentPopover from "../EditComponentPopover.svelte"
+  import { Icon } from "@budibase/bbui"
+  import { store } from "builderStore"
+  import { runtimeToReadableBinding } from "builderStore/dataBinding"
+  import { isJSBinding } from "@budibase/string-templates"
+
+  export let item
+  export let componentBindings
+  export let bindings
+  export let anchor
+  export let hideToggle = false
+
+  const getReadableText = () => {
+    if (item.label) {
+      return isJSBinding(item.label)
+        ? "(JavaScript function)"
+        : runtimeToReadableBinding([...bindings, componentBindings], item.label)
+    }
+    return item.field
+  }
+
+  const parseSettings = settings => {
+    return settings
+      .filter(setting => setting.key !== "field")
+      .map(setting => {
+        return { ...setting, nested: true }
+      })
+  }
+
+  $: readableText = getReadableText(item)
+  $: componentDef = store.actions.components.getDefinition(item._component)
+</script>
+
+<div class="list-item-body">
+  <div class="list-item-left">
+    <EditComponentPopover
+      {anchor}
+      componentInstance={item}
+      {componentBindings}
+      {bindings}
+      {parseSettings}
+      on:change
+    >
+      <div slot="header" class="type-icon">
+        <Icon name={componentDef.icon} />
+        <span>{item.field}</span>
+      </div>
+    </EditComponentPopover>
+    <div class="field-label">{readableText}</div>
+  </div>
+  <div
```

Comment:
our tooltip component can't be made to align left, so it kept going off the screen. I just opted to use a html `title` instead.

Reply:
Would much prefer to use this, but I was having issues with the length of the text + how the bounding rect in the tooltip component is calculated, any advice? 
<img width="283" alt="Screenshot 2023-11-08 at 09 04 46" src="https://github.com/Budibase/budibase/assets/12501626/aa5b71b6-59d9-493c-9c31-c9d8a4caa89f">